modules/library/main/src/main/java/org/geotools/styling/DefaultResourceLocator.java

```diff
@@ -71,6 +72,23 @@ public URL locateResource(String uri) {
         return url;
     }
 
+    /**
+     * Variant to URL.getQuery() that won't balk at having # in the query (e.g., a color in a
+     * parametric SVG reference)
+     *
+     * @param url
+     * @return
+     */
+    private String getQueryPart(URL url) {
+        String external = url.toExternalForm();
+        int idx = external.indexOf('?');
+        if (idx == -1 || idx == external.length() - 1) {
+            return null;
+        } else {
```

Comment:
Unnecessary ``else`` it can return directly.

Reply:
Unless covered by the style guide, this comments are a matter of personal taste and could be skipped (we won't get agreement on this unless by writing an addendum to the Google Style, but then, how does one enforce it?).

Reply:
I agree ... I never thought about this as a styling issue, I didn't know you were doing it on propose, my bad.